forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/services/openbanking/event/EventResponseUtil.java

```diff
@@ -112,10 +117,34 @@ private OBCallbackUrlResponseData1 toOBCallbackUrlResponseData1(FRCallbackUrl fr
     }
 
     private Links toSelfLink(FRCallbackUrl frCallbackUrl) {
-        return this.resourceLinkService.toSelfLink(frCallbackUrl, discovery -> discovery.getVersion(version).getGetCallbackUrls());
+        return this.resourceLinkService.toSelfLink(frCallbackUrl, getUrlCallbackFunction());
     }
 
     private boolean shouldHaveMetaSection() {
         return haveMetaSection && version != null;
     }
+
+    public Function<DiscoveryConfigurationProperties.EventNotificationApis, String> getUrlCallbackFunction() {
```

Comment:
Minor - can we remove `Function` from the end to make it consistent with the other APIs?

Reply:
We can do it but means that we need apply a lot of changes around all controllers that use it. no time...